runtime/mock/src/circuit_config.rs

```diff
@@ -1,18 +1,118 @@
 use crate::*;
 
-use frame_support::{parameter_types, traits::ConstU32, PalletId};
+use sp_std::vec;
+
+use frame_support::{ensure, parameter_types, traits::ConstU32, weights::Weight, PalletId};
 use pallet_grandpa_finality_verifier::bridges::runtime as bp_runtime;
 use sp_core::H256;
-
+use sp_runtime::{
+    traits::{BlakeTwo256, Convert, One},
+    Perbill,
+};
 impl t3rn_primitives::EscrowTrait<Runtime> for Runtime {
     type Currency = Balances;
     type Time = Timestamp;
 }
 
+pub struct GlobalOnInitQueues;
+
+impl pallet_clock::traits::OnHookQueues<Runtime> for GlobalOnInitQueues {
+    fn process(n: BlockNumber, on_init_weight_limit: Weight) -> Weight {
+        let mut weights_consumed = vec![];
+        const PROCESS_SIGNAL_SHARE: u32 = 15;
+        const XTX_TICK_SHARE: u32 = 35;
+        const REVERT_XTX_SHARE: u32 = 35;
+        const BUMP_ROUND_SHARE: u32 = 5;
+        const CALC_CLAIMABLE_SHARE: u32 = 10;
+        if PROCESS_SIGNAL_SHARE
+            + XTX_TICK_SHARE
+            + REVERT_XTX_SHARE
+            + BUMP_ROUND_SHARE
+            + CALC_CLAIMABLE_SHARE
+            > 100
+        {
+            log::error!(
+                "GlobalOnInitQueues::Invalid shares exceed 100%, returning 0 - re-check the shares"
+            );
+            return 0
+        }
+        // Iterate over all pre-init hooks implemented by pallets and return aggregated weight
+        weights_consumed.push(Circuit::process_signal_queue(
+            n,
+            BlockNumber::one(),
+            Perbill::from_percent(PROCESS_SIGNAL_SHARE) * on_init_weight_limit,
```

Comment:
Should this be safe? Like `Perbill::from_percent(PROCESS_SIGNAL_SHARE).checked_mul(on_init_weight_limit).unwrap()`

Reply:
unwrap isn't safe, but feel free to change to .saturated_mul

Reply:
On it!